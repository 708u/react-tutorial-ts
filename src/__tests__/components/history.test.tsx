import React from 'react';
import renderer from 'react-test-renderer';
import History, { History as HistoryType } from 'components/history';

describe('history component test', () => {
  it('should be same as a previous snapshot.', () => {
    const history: HistoryType = [
      { squares: [null, null, null, null, null, null, null, null, null] },
      {
        squares: ['O', null, null, null, null, null, null, null, null],
        position: { row: 1, col: 1 },
      },
      {
        squares: ['O', 'X', null, null, null, null, null, null, null],
        position: { row: 1, col: 2 },
      },
    ];
    const component = renderer.create(<History history={history} current={0} onClick={() => () => true} />);
    expect(component.toJSON()).toMatchSnapshot();

    // determine if current has applied.
    const secondComponent = renderer.create(<History history={history} current={2} onClick={() => () => true} />);

    expect(secondComponent.toJSON()).toMatchSnapshot();
  });
});
