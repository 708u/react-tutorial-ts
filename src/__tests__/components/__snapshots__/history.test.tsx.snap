// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`history component test should be same as a previous snapshot. 1`] = `
<ol
  className="sc-bdfBwQ dDRKCN"
>
  <li>
    <button
      className="btn-bold"
      onClick={[Function]}
      type="button"
    >
      Go to game start
    </button>
  </li>
  <li>
    <button
      className=""
      onClick={[Function]}
      type="button"
    >
      Go to move #1. [row: 1, col: 1]
    </button>
  </li>
  <li>
    <button
      className=""
      onClick={[Function]}
      type="button"
    >
      Go to move #2. [row: 1, col: 2]
    </button>
  </li>
</ol>
`;

exports[`history component test should be same as a previous snapshot. 2`] = `
<ol
  className="sc-gsTCUz LxBZ"
>
  <li>
    <button
      className=""
      onClick={[Function]}
      type="button"
    >
      Go to game start
    </button>
  </li>
  <li>
    <button
      className=""
      onClick={[Function]}
      type="button"
    >
      Go to move #1. [row: 1, col: 1]
    </button>
  </li>
  <li>
    <button
      className="btn-bold"
      onClick={[Function]}
      type="button"
    >
      Go to move #2. [row: 1, col: 2]
    </button>
  </li>
</ol>
`;
